### PR TITLE
mds: Reorganize class members in MDLog header

### PR DESCRIPTION
**Pull request by varshar16**
Fixes: https://tracker.ceph.com/issues/42192
Signed-off-by: Varsha Rao <varao@redhat.com>

<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by varshar16** [comment]
jenkins retest this please

File: src/mds/MDLog.h [hunk 1]
```diff
@@ -42,181 +40,41 @@ enum {
...
 #include "common/Cond.h"
+#include "common/Finisher.h"
+#include "common/Thread.h"
 
 #include "LogSegment.h"
 
 #include <list>
+#include <map>
+
+using std::map;
```
Comment (batrick) [inline src/mds/MDLog.h]: Let's move this into `MDLog`.

File: src/mds/MDLog.h [hunk 2]
```diff
@@ -332,6 +171,141 @@ class MDLog {
...
+  int unflushed = 0;
+  bool capped = false;
+
+  // Log position which is persistent *and* for which
+  // submit_entry wait_for_safe callbacks have already
+  // been called.
+  uint64_t safe_pos = 0;
+
+  inodeno_t ino;
+  Journaler *journaler = 0;
```
Comment (batrick) [inline src/mds/MDLog.h]: `nullptr`

File: src/mds/MDLog.h [hunk 3]
```diff
@@ -332,6 +171,141 @@ class MDLog {
...
+
+  PerfCounters *logger = 0;
```
Comment (batrick) [inline src/mds/MDLog.h]: `nullptr`

File: src/mds/MDLog.h [hunk 4]
```diff
@@ -332,6 +171,141 @@ class MDLog {
...
+  void _journal_segment_subtree_map(MDSContext *onsync);
+
+  void try_expire(LogSegment *ls, int op_prio);
+  void _maybe_expired(LogSegment *ls, int op_prio);
+  void _expired(LogSegment *ls);
+  void _trim_expired_segments();
+  void write_head(MDSContext *onfinish);
+
+  // -- events --
+  LogEvent *cur_event = NULL;
```
Comment (batrick) [inline src/mds/MDLog.h]: `nullptr`

**Comment by varshar16** [comment]
jenkins retest this please

**Pull request merged**